tests/aws/services/sns/test_sns.py

```diff
@@ -822,6 +822,45 @@ def test_list_subscriptions(
 
         assert all((sub["TopicArn"], sub["Endpoint"]) in sorting_list for sub in all_subs)
 
+    @markers.aws.validated
+    def test_list_subscriptions_by_topic_pagination(
+        self, sns_create_topic, sns_subscription, snapshot, aws_client
+    ):
+        # ordering of the listing seems to be not consistent, so we can transform its value
+        snapshot.add_transformers_list(
+            [
+                snapshot.transform.key_value("Endpoint"),
+                snapshot.transform.key_value("NextToken"),
+            ]
+        )
+
+        base_phone_number = "+12312312"
+        topic_arn = sns_create_topic()["TopicArn"]
+        for phone_suffix in range(101):
+            phone_number = f"{base_phone_number}{phone_suffix}"
+            sns_subscription(TopicArn=topic_arn, Protocol="sms", Endpoint=phone_number)
+
+        response = aws_client.sns.list_subscriptions_by_topic(TopicArn=topic_arn)
+        page_1_subs = {sub["SubscriptionArn"] for sub in response["Subscriptions"]}
+        # not snapshotting the results, it contains 100 entries
+        assert "NextToken" in response
+        # seems to be b64 encoded
+        assert response["NextToken"].endswith("==")
+        assert len(response["Subscriptions"]) == 100
```

Comment:
Wouldn't you want the `len(page_1_subs)`? As I assumed you made it a set to catch dedups? And you are asserting only one should come on page two.

Reply:
I think it's similar, I was using a set above not for deduplication but just because the `in` call would be a tiny bit faster 😅 but yes, calling `len` would be the same on both. 
The set was made is just to verify that the one coming on page 2 is not already present in page 1. But I'll update it to be a bit clearer! Thanks!

Reply:
haaa! Now I understand. Just my assumption that was wrong :stuck_out_tongue_closed_eyes: